crates/router/src/connector/square/transformers.rs

```diff
@@ -241,7 +241,7 @@ impl TryFrom<&types::PaymentsAuthorizeRouterData> for SquarePaymentsRequest {
             api::PaymentMethodData::Card(_) => {
                 let pm_token = item.get_payment_method_token()?;
                 Ok(Self {
-                    idempotency_key: Secret::new(item.attempt_id.clone()),
+                    idempotency_key: Secret::new(item.connector_request_reference_id.clone()),
```

Comment:
Hello @Ln11211 ,

According api docs of square , the merchant reference id should be `reference_id` not `idempotency_key` . Could you please replace `idempotency_key`  with `reference_id` ? 

Please find api documentation for square [here](https://developer.squareup.com/reference/square/payments-api/create-payment)

Reply:
Yes I'm sorry I overlooked that part. I made the changes and raised a new PR.